### PR TITLE
Opset-11 support (negative axis) for reduce ops

### PR DESCRIPTION
**Pull request by shahasad**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/cpu_execution_provider.cc [hunk 1]
```diff
@@ -318,6 +318,32 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain,
...
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, float, ReduceMean);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, int32_t, ReduceMean);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, float, ReduceMin);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, int32_t, ReduceMin);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, int64_t, ReduceMin);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, float, ReduceProd);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, int32_t, ReduceProd);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, float, ReduceSum);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, double, ReduceSum);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 11, int32_t, ReduceSum);
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/cpu_execution_provider.cc]: Could you please enable the test in the execution frame tests that was disabled in #1868 on account of the missing opset-11 ReduceSum kernel ?

File: onnxruntime/test/providers/cpu/reduction/reduction_test_cases.inl [hunk 2]
```diff
@@ -4,5374 +4,6180 @@
...
+	#pragma clang optimize off
 #else
-    #pragma GCC push_options
-    #pragma GCC optimize ("O0")
+	#pragma GCC push_options
+	#pragma GCC optimize ("O0")
 #endif
 #endif
 ReductionTestCases testcases = {
-    // input_data
```
Comment (hariharans29) [inline onnxruntime/test/providers/cpu/reduction/reduction_test_cases.inl]: is this large diff just formatting?

File: onnxruntime/test/onnx/main.cc [hunk 3]
```diff
@@ -446,25 +446,6 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
       {"sequence_model2", "SequenceConstruct not implemented yet"},
       {"sequence_model1", "Sequence* not implemented yet"},
       {"scatter_elements_with_negative_indices", "ScatterElements(11) not implemented yet"},
-      {"reduce_sum_square_negative_axes_keepdims_random", "ReduceSumSquare(11) not implemented yet"},
```
Comment (hariharans29) [inline onnxruntime/test/onnx/main.cc]: Could you remove relevant tests from onnx_backend_test_series.py as well ?

File: onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc [hunk 4]
```diff
@@ -57,11 +57,15 @@ TEST(ReductionOpTest, ReductionVariationTest) {
       std::vector<int64_t> expected_values;
       for (auto v : std::get<2>(a.second))
         expected_values.push_back(static_cast<int64_t>(v));
-      TestReduceOp<int64_t>(a.first, input_dims, input_data, attributes.axes_, attributes.keep_dims_,
+      TestReduceOp<int64_t>(a.first, 7, input_dims, input_data, attributes.axes_, attributes.keep_dims_,
```
Comment (hariharans29) [inline onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc]: It looks like NGraph pieplines are breaking on account of this. Maybe they have not been excluded for the negative axes case ? Line 28 looks like a suspect - 

"if (op.compare("ArgMax") == 0 || op.compare("ArgMin") == 0) {"

Should it have something like Reduce* ?

File: onnxruntime/test/providers/cpu/reduction/reduction_test_cases.inl (hunk 2 above)
Comment (shahasad) [reply onnxruntime/test/providers/cpu/reduction/reduction_test_cases.inl]: This .inl file is auto generated from the python script. I think the formatting changes came from the 't' to '\t' change in line 81 there.

**Comment by skottmckay** [comment]
>     for (auto i : axes_) {

axes.reserve(axes_.size());

---
Refers to: onnxruntime/core/providers/nuphar/compiler/x86/op_ir_creator/math/reduce_ops.cc:122 in 3f91286. [](commit_id = 3f91286d2babcbb036ec53e47c4fe201169797fa, deletion_comment = False)

**Pull request merged**